lib/modules/manager/cargo/artifacts.ts

```diff
@@ -64,11 +104,21 @@ export async function updateArtifacts({
   try {
     await writeLocalFile(packageFileName, newPackageFileContent);
     logger.debug('Updating ' + lockFileName);
-    await cargoUpdate(
-      packageFileName,
-      isLockFileMaintenance,
-      config.constraints?.rust
-    );
+
+    if (isLockFileMaintenance) {
+      await cargoUpdate(packageFileName, true, config.constraints?.rust);
+    } else if (updatedDeps.find((dep) => dep.lockedVersion)) {
```

Comment:
what if there are only some updates with a locked version? can that happen? they would now be silently ignored

Reply:
The only case I can think of is if `Cargo.lock` file is not in sync with `Cargo.toml`. Imo this is fine as it's a user mistake and will be corrected the next time the lock file is touched by cargo.

And I can't really think of a good way to handle this anyway. If I regenerate the lock file before trying to precisely update dependencies, then the `lockedVersion` of other dependencies will no longer match the state in `Cargo.lock` an we'll have no way to reference dependencies for a precise update.

Reply:
we should probably return an artifact error If not all updates have a locked version. or we do a full lockfile update as we do currently and only log a warning.

Reply:
Decided to do a regular workspace update and log a warning.